File: src/test/java/de/pflugradts/passbird/domain/service/PasswordProviderFaker.java
package de.pflugradts.passbird.domain.service;

import de.pflugradts.passbird.domain.model.password.PasswordRequirements;
import de.pflugradts.passbird.domain.model.transfer.Bytes;
import de.pflugradts.passbird.domain.service.password.provider.PasswordProvider;
import lombok.AccessLevel;
import lombok.RequiredArgsConstructor;
import static org.mockito.ArgumentMatchers.any;
import static org.mockito.BDDMockito.given;

@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
public class PasswordProviderFaker {

    private PasswordProvider passwordProvider;

    public static PasswordProviderFaker faker() {
        return new PasswordProviderFaker();
    }

    public PasswordProviderFaker forInstance(final PasswordProvider passwordProvider) {
        this.passwordProvider = passwordProvider;
        return this;
    }

    public PasswordProviderFaker withCreatingThisPassword(final Bytes password) {
        given(passwordProvider.createNewPassword(any(PasswordRequirements.class))).willReturn(password);
        return this;
    }

    public PasswordProvider fake() {
        return passwordProvider;
    }

}

File: src/test/kotlin/de/pflugradts/passbird/domain/service/password/provider/PasswordProviderFaker.kt
package de.pflugradts.passbird.domain.service.password.provider

import de.pflugradts.passbird.domain.model.shell.Shell
import de.pflugradts.passbird.domain.model.shell.Shell.Companion.shellOf
import io.mockk.every

fun fakePasswordProvider(
    instance: PasswordProvider,
    withCreatedPassword: Shell = shellOf("password"),
) {
    every { instance.createNewPassword(any()) } returns withCreatedPassword
}
